### PR TITLE
Task yc/j57 add user menu on hover

### PR DESCRIPTION
**Pull request by EugeniusGR**
image

### CODE REVIEW

File: frontend/src/components/common/header/header.tsx [hunk 1]
```diff
@@ -40,6 +50,18 @@ const Header: React.FC = () => {
                   alt="avatar"
                   loading="lazy"
                 />
+                {isVisible ? (
```
Comment (ghost) [inline frontend/src/components/common/header/header.tsx]:
```suggestion
                {isVisible && (
```

File: frontend/src/hooks/app-click-selector/app-click-selector.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
```
Comment (ghost) [inline frontend/src/hooks/app-click-selector/app-click-selector.ts]:
```suggestion
```

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by Sayonarko — DISMISSED** [review]

File: frontend/src/components/common/header/styles.module.scss [hunk 3]
```diff
@@ -61,3 +62,44 @@
...
+  box-shadow: 0px 5px 10px var(--color-primary-dark);
+}
+
+.dropDownListElement {
+  background-color: var(--color-header-background);
+  color: #fff;
+  padding: 10px 5px;
+  border-radius: 5px;
+  margin: 5px 0;
+  transition: all 0.3s ease;
```
Comment (Denidzio) [inline frontend/src/components/common/header/styles.module.scss:108]: We can also add 'cursor: pointer' for better UX;

**Review by Denidzio — DISMISSED** [review]

**Comment by EugeniusGR** [comment]
[Add user menu on click by avatar](https://trello.com/c/GlR5MyT7/57-add-user-menu-on-click-by-avatar)

File: frontend/src/components/common/header/header.tsx [hunk 4]
```diff
@@ -13,6 +13,12 @@ const Header: React.FC = () => {
 
   const hasUser = Boolean(user);
 
+  const { ref, isVisible, setIsVisible } = useVisible(false);
+
+  const handleMenutoggle = (): void => {
```
Comment (yevhenprybytok) [inline frontend/src/components/common/header/header.tsx]:
```suggestion
  const handleMenuToggle = (): void => {
```

File: frontend/src/components/common/header/header.tsx [hunk 5]
```diff
@@ -31,7 +37,11 @@ const Header: React.FC = () => {
             </ul>
             <div className={styles.userInfo}>
               <Button label="+ Create Podcast" type={ButtonType.BUTTON} />
-              <div className={styles.profile}>
+              <div
+                className={styles.profile}
+                ref={ref}
+                onClick={(): void => handleMenutoggle()}
```
Comment (yevhenprybytok) [inline frontend/src/components/common/header/header.tsx]:
```suggestion
                onClick={handleMenutoggle}
```

File: frontend/src/components/common/header/header.tsx [hunk 6]
```diff
@@ -40,6 +50,16 @@ const Header: React.FC = () => {
                   alt="avatar"
                   loading="lazy"
                 />
+                {isVisible && (
+                  <div className={styles.dropDown}>
```
Comment (yevhenprybytok) [inline frontend/src/components/common/header/header.tsx:55]: Does this work with the animation (in-out)?

File: frontend/src/components/common/header/styles.module.scss [hunk 7]
```diff
@@ -61,3 +63,45 @@
   font-size: 18px;
   line-height: 21px;
 }
+
+.dropDown {
+  position: relative;
+}
+
+.dropDownList {
```
Comment (yevhenprybytok) [inline frontend/src/components/common/header/styles.module.scss:76]: Nest classes depending on their positioning

File: frontend/src/hooks/app-click-selector/app-click-selector.ts [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+type VisibleHookType = {
+  ref: React.RefObject<HTMLDivElement>;
+  isVisible: boolean;
+  setIsVisible: React.Dispatch<React.SetStateAction<boolean>>;
+};
+
+const useVisible = (isInitialIsVisible: boolean): VisibleHookType => {
+  const [isVisible, setIsVisible] = useState<boolean>(isInitialIsVisible);
+  const ref = useRef<HTMLDivElement>(null);
+  const handleClickOutside = (evt: MouseEvent): void => {
```
Comment (yevhenprybytok) [inline frontend/src/hooks/app-click-selector/app-click-selector.ts:13]:
```suggestion
  
  const handleClickOutside = (evt: MouseEvent): void => {
```

File: frontend/src/hooks/app-click-selector/app-click-selector.ts [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+    const element = evt.target as HTMLElement;
+    if (ref.current && !ref.current.contains(element)) {
+      setIsVisible(false);
+    }
+  };
+  useEffect(() => {
```
Comment (yevhenprybytok) [inline frontend/src/hooks/app-click-selector/app-click-selector.ts:20]:
```suggestion
  
  useEffect(() => {
```

File: frontend/src/hooks/app-click-selector/app-click-selector.ts [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+    document.addEventListener('click', handleClickOutside, {
+      capture: true,
+    });
+    return (): void => {
```
Comment (yevhenprybytok) [inline frontend/src/hooks/app-click-selector/app-click-selector.ts:25]:
```suggestion
    
    return (): void => {
```

File: frontend/src/hooks/app-click-selector/app-click-selector.ts [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+      document.removeEventListener('click', handleClickOutside, {
+        capture: true,
+      });
+    };
+  }, []);
+  return { ref, isVisible, setIsVisible };
```
Comment (yevhenprybytok) [inline frontend/src/hooks/app-click-selector/app-click-selector.ts:32]:
```suggestion
  
  return { ref, isVisible, setIsVisible };
```

**Review by yevhenprybytok — CHANGES_REQUESTED** [review]

File: frontend/src/components/common/header/header.tsx [hunk 12]
```diff
@@ -31,7 +37,11 @@ const Header: React.FC = () => {
             </ul>
             <div className={styles.userInfo}>
               <Button label="+ Create Podcast" type={ButtonType.BUTTON} />
-              <div className={styles.profile}>
+              <div
```
Comment (ghost) [inline frontend/src/components/common/header/header.tsx]: remove `onClick` on div

File: frontend/src/components/common/header/header.tsx [hunk 13]
```diff
@@ -40,6 +50,16 @@ const Header: React.FC = () => {
                   alt="avatar"
                   loading="lazy"
                 />
+                {isVisible && (
+                  <div className={styles.dropDown}>
+                    <div className={styles.dropDownList}>
+                      <div className={styles.dropDownListElement}>
+                        + Add Podcast
+                      </div>
```
Comment (ghost) [inline frontend/src/components/common/header/header.tsx]: this it link, right?

also, make it as `ul` with `li` link-items

File: frontend/src/components/common/header/styles.module.scss [hunk 14]
```diff
@@ -61,3 +63,45 @@
...
+}
+
+.dropDownListElement {
+  background-color: var(--color-header-background);
+  color: #fff;
+  padding: 10px 5px;
+  border-radius: 5px;
+  margin: 5px 0;
+  transition: all 0.3s ease;
+  cursor: pointer;
```
Comment (ghost) [inline frontend/src/components/common/header/styles.module.scss]:
```suggestion
```

**Review by ghost — CHANGES_REQUESTED** [review]

File: frontend/src/components/common/header/header.tsx (hunk 6 above)
Comment (EugeniusGR) [reply frontend/src/components/common/header/header.tsx:55]: I found how to make animations, but its needs a _'react-transition-group'_ lib

File: frontend/src/components/common/header/header.tsx [hunk 15]
```diff
@@ -31,15 +37,32 @@ const Header: React.FC = () => {
...
+                  <img
+                    className={styles.profileAvatar}
+                    src={defaultAvatar}
+                    width="40px"
+                    height="40px"
+                    alt="avatar"
+                    loading="lazy"
+                  />
+                  {isVisible && (
+                    <div className={styles.dropDown}>
```
Comment (ghost) [inline frontend/src/components/common/header/header.tsx]: div can't be nested in button

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**